lib/ansible/constants.py

```diff
@@ -36,6 +40,26 @@ def _deprecated(msg, version):
         sys.stderr.write(' [DEPRECATED] %s, to be removed in %s\n' % (msg, version))
 
 
+def handle_config_noise(display=None):
+
+    if display is not None:
+        w = display.warning
+        d = display.deprecated
+    else:
+        w = _warning
+        d = _deprecated
+
+    while config.WARNINGS:
+        warn = config.WARNINGS.pop()
```

Comment:
Minor nit, but aren't we displaying these in the opposite order they arrived in?

```suggestion
        warn = config.WARNINGS.pop(0)
```

The accumulators could also be changed to deques (from their current list + set) to make that more efficient with O(1) `popleft()`. Since we're emptying them so frequently, I'm not sure there's any real benefit to config doing its own warning de-dupe anyway, especially if we ditch the pre-display fallback (since it already does warning de-dupe).

Reply:
good catch